src/gurt/debug.c

```diff
@@ -592,6 +602,8 @@ d_log_init(void)
 
 	d_log_sync_mask();
 out:
+	if (buffer != NULL)
```

Comment:
just D_FREE(buffer).

free() itself allows free(NULL), and D_FREE checks for this as well.  We reviewed the code a while ago and decided that the shorter form is easier to read, and the extra log messages actually make it easier to tell the progression of the application from the log file.  An exception is where free is called in a for loop where most of the entries are NULL, any we typically do check in those cases to avoid spamming the log file with duplicate entries.

Reply:
okay